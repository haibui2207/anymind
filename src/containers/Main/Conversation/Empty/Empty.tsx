import React, { memo } from "react";

import coverImage from "../../../../assets/images/empty-conversation.jpg";

import useStyles from "./empty.styles";

const Empty: React.FC<{}> = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <img
        alt="No message"
        className={classes.image}
        src={coverImage}
        data-test="empty-image"
      />
      <h2 className={classes.title} data-test="empty-title">
        No message found
      </h2>
      <p className={classes.description} data-test="empty-description">
        Let's start your conversation
      </p>
    </div>
  );
};

export default memo(Empty);
