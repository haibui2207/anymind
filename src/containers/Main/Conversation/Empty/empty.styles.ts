import { createUseStyles } from "react-jss";

import { rem } from "../../../../utils/jss";

export default createUseStyles({
  root: {
    padding: rem(16),
    width: "100%",
    height: "100%",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "column",
    boxSizing: "border-box",
    textAlign: "center",
  },
  image: { display: "block", width: "100%", maxWidth: rem(250) },
  title: {
    marginTop: rem(16),
    marginBottom: 0,
    fontSize: rem(32),
    lineHeight: rem(48),
  },
  description: {
    marginTop: rem(12),
    marginBottom: 0,
    fontSize: rem(16),
    lineHeight: rem(24),
  },
});
