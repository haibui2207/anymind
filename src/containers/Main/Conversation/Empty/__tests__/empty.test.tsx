import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import Empty from "../Empty";

describe("Testing Main > Conversation > Empty", (): void => {
  let wrapper: any;

  beforeAll((): void => {
    wrapper = mount(<Empty />);
  });

  it("should render match snapshot", (): void => {
    const wrapper = shallow(<Empty />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should have an image", (): void => {
    expect(wrapper.find('[data-test="empty-image"]')).toHaveLength(1);
  });

  it("should have a title", (): void => {
    expect(wrapper.find('[data-test="empty-title"]')).toHaveLength(1);
  });

  it("should have a description", (): void => {
    expect(wrapper.find('[data-test="empty-description"]')).toHaveLength(1);
  });
});
