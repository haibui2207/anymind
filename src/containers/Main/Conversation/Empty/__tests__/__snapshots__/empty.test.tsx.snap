// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Main > Conversation > Empty should render match snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <img
    alt="No message"
    className="image-0-2-2"
    data-test="empty-image"
    src="empty-conversation.jpg"
  />
  <h2
    className="title-0-2-3"
    data-test="empty-title"
  >
    No message found
  </h2>
  <p
    className="description-0-2-4"
    data-test="empty-description"
  >
    Let's start your conversation
  </p>
</div>
`;
